### PR TITLE
Add $lib.storm.run() function

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: changes/9b0c3998325687c0ac92251ed3b9ddbf.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+desc: Added $lib.storm.run() to programatically invoke Storm.
```
Comment (Cisphyx) [inline changes/9b0c3998325687c0ac92251ed3b9ddbf.yaml:2]:
```suggestion
desc: Added $lib.storm.run() to programmatically invoke Storm.
```

File: synapse/lib/stormlib/storm.py [hunk 2]
```diff
@@ -101,15 +107,43 @@ class LibStorm(s_stormtypes.Lib):
                       {'name': 'text', 'type': 'str', 'desc': 'A storm expression string.'},
                       {'name': 'cast', 'type': 'str', 'desc': 'A type to cast the result to.', 'default': None},
                   ),
-                  'returns': {'type': 'any', 'desc': 'The value of the expression and optional cast.', }}},
+                  'returns': {'type': 'any', 'desc': 'The value of the expression and optional cast.'}}},
+        {'name': 'run', 'desc': rundesc,
+         'type': {'type': 'function', '_funcname': '_runStorm',
+                  'args': (
+                      {'name': 'query', 'type': 'str', 'desc': 'A storm query string.'},
```
Comment (Cisphyx) [inline synapse/lib/stormlib/storm.py:114]:
```suggestion
                      {'name': 'query', 'type': 'str', 'desc': 'A Storm query string.'},
```

File: synapse/lib/stormlib/storm.py [hunk 3]
```diff
@@ -9,8 +9,16 @@
 evaldesc = '''\
 Evaluate a storm runtime value and optionally cast/coerce it.
 
-NOTE: If storm logging is enabled, the expression being evaluated will be logged
-separately.
+Note:
+    If storm logging is enabled, the expression being evaluated will be logged
```
Comment (Cisphyx) [inline synapse/lib/stormlib/storm.py:13]:
```suggestion
    If Storm logging is enabled, the expression being evaluated will be logged
```

File: synapse/lib/stormlib/storm.py [hunk 4]
```diff
@@ -101,15 +109,43 @@ class LibStorm(s_stormtypes.Lib):
...
             'eval': self._evalStorm,
         }
 
+    async def _runStorm(self, query, opts=None):
+
+        opts = await s_stormtypes.toprim(opts)
+        query = await s_stormtypes.tostr(query)
+
+        if opts is None:
+            opts = {}
```
[the remark this answers is not in the document]
Comment (invisig0th) [reply synapse/lib/stormlib/storm.py:135]: No, they can specify them :+1:

File: synapse/lib/stormlib/storm.py [hunk 5]
```diff
@@ -9,8 +9,16 @@
...
 
-NOTE: If storm logging is enabled, the expression being evaluated will be logged
-separately.
+Note:
+    If Storm logging is enabled, the expression being evaluated will be logged
+    separately.
+'''
+
+rundesc = '''
+Run a storm query and yield the messages output by the Storm interpreter.
```
Comment (Cisphyx) [inline synapse/lib/stormlib/storm.py:18]:
```suggestion
Run a Storm query and yield the messages output by the Storm interpreter.
```

**Review by Cisphyx — DISMISSED** [review]

**Review by vEpiphyte — APPROVED** [review]

**Pull request merged**